tests/countries/test_sint_maarten.py

```diff
@@ -0,0 +1,190 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date, timedelta
+from unittest import TestCase
+
+from holidays.countries.sint_maarten import SintMaarten, SX, SXM
+from tests.common import CommonCountryTests
+
+
+class TestSintMaarten(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(SintMaarten, years=range(2011, 2077))
+
+    def test_country_aliases(self):
+        self.assertAliases(SintMaarten, SX, SXM)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(SintMaarten(years=2010))
+
+    def test_2017(self):
+        self.assertHolidays(
+            SintMaarten(years=2017),
+            ("2017-01-01", "New Year's Day"),
+            ("2017-04-14", "Good Friday"),
+            ("2017-04-16", "Easter Sunday"),
+            ("2017-04-17", "Easter Monday"),
+            ("2017-04-27", "King's Day"),
+            ("2017-05-02", "Carnival Day"),
+            ("2017-05-01", "Labour Day"),
+            ("2017-05-25", "Ascension Day"),
+            ("2017-06-04", "Whit Sunday"),
+            ("2017-07-01", "Emancipation Day"),
+            ("2017-10-09", "Constitution Day"),
+            ("2017-11-01", "All Saints' Day"),
+            ("2017-11-11", "Sint Maarten Day"),
+            ("2017-12-15", "Kingdom Day"),
+            ("2017-12-25", "Christmas Day"),
+            ("2017-12-26", "Second day of Christmas"),
+        )
+
+    def test_pre_2011_branch_coverage(self):
+        class TestSintMaarten(SintMaarten):
+            start_year = 2010
+
+        holidays = TestSintMaarten(years=2010)
+        # Verify holidays that should exist
+        self.assertIn(date(2010, 1, 1), holidays)  # New Year's Day
```

Comment:
We don't operate w/ classic `date()` objects in tests and use framework specific assertions instead. Please refactor (see other tests examples).